raiden-ts/src/config.ts

```diff
@@ -0,0 +1,32 @@
+import * as t from 'io-ts';
+
+/**
+ * An Raiden configuration object to overwrite the Raiden default configuration.
+ * 
+ * @matrixServer Matrix server to initliaze the [[Raiden]] instance with.
+ * The matrix server can't be changed later on.
+ */
+export const PartialRaidenConfig = t.partial({
+  matrixServer: t.string,
+});
+
+/**
+ * A Raiden configuration object with required parameters and
+ * optional parameters from [[PartialRaidenConfig]].
+ * 
+ * @matrixServerLookup Matrix server URL to fetch existing matrix servers from. 
+ * After intializing a [[Raiden]] instance, the matrix server can't be changed later on.
+ * @revealTimeout Timeout for secrets to be revealed
+ * @settleTimeout Timeout for channels to be settled
+ */
+export const RaidenConfig = t.intersection([
+  t.type({
+    matrixServerLookup: t.string,
+    revealTimeout: t.number,
+    settleTimeout: t.number,
+  }),
+  PartialRaidenConfig,
+]);
+
+export type RaidenConfigDefaults = t.TypeOf<typeof RaidenConfig>;
```

Comment:
Please, make it an empty interface inheriting from the type (to not fall into [this pitfall](https://github.com/raiden-network/light-client/wiki/SDK-Development#a-note-about-io-ts-and-type-complexity)), and rename it to the same name of the codec (`RaidenConfig`) for consistency with the rest of the code which uses this pattern.